doc/xgboost_on_sqlflow.md

```diff
@@ -0,0 +1,139 @@
+# xgboost on sqlflow
+
+## XGBoost 
+
+Gradient boosting machine (GBM) is a widely used (supervised) machine learning method, 
+which trains a bunch of weak learners, typically decision trees, 
+in a gradual, additive and sequential manner. 
+A lot of winning solutions of data mining and machine learning challenges, 
+such as : Kaggle, KDD cup, are based on GBM or related techniques.
+
+XGBoost (https://xgboost.ai/) is an optimized distributed gradient boosting library designed to be highly efficient, 
+flexible and portable, which is often regarded as one of the best GBM frameworks.
+
+
+## xgboost on sqlflow via ant-xgboost
+   
+### Overview
+
+We use [ant-xgboost](https://github.com/alipay/ant-xgboost) as backend,
+which is forked from [xgboost](https://github.com/dmlc/xgboost).
+_Ant-xgboost_ is nearly same as original xgboost, 
+except some improvements to make xgboost easier to use, 
+such as better early stopping strategy, parameter checking, and end to end launcher.
+
+### User Experience
+    
+In terms of sqlflow users, _xgboost_ is an alternative _Estimator_ like _Tensorflow Estimators_. 
+Working with xgboost is quite similar to working with Tensorflow estimators; 
+just change `TRAIN DNNClassifier` into `TRAIN XGBoostEstimator`. 
```

Comment:
Currently, SQLFlow using the env `SQLFlow_submittor` to support the different submitters(`alps` for ALPS lib or Tensorflow Keras by default): 

https://github.com/sql-machine-learning/sqlflow/blob/317ef496f36f5412f47d9f56ba894d590844bfa4/sql/executor.go#L223-L225

Maybe we can follow this logic instead of using the prefix of the model name. In addition, SQLFlow will support the different backend by passing  the `submitter` field in the Session struct: https://github.com/sql-machine-learning/sqlflow/issues/539